### PR TITLE
Add the year range slider for filtering papers by published year

### PR DESCRIPTION
**Pull request by TylorShin**

### CODE REVIEW

File: app/containers/filterContainer/yearRangeSlider/index.tsx [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+        const yearSet = props.yearInfo.find(ys => ys.year === MIN_YEAR + i);
+        return {
+          year: MIN_YEAR + i,
+          docCount: yearSet ? yearSet.docCount : 0,
+        };
+      }),
+    [props.yearInfo]
+  );
+
+  const qp: SearchPageQueryParams = getQueryParamsObject(props.location.search);
```
Comment (NohYujeong) [inline app/containers/filterContainer/yearRangeSlider/index.tsx]: `props.location.search`

^ This variable seems to be used frequently. Why not declare it with a simpler variable name?

File: app/containers/filterContainer/yearRangeSlider/index.tsx [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+  return (
+    <>
+      {props.filteredYearSet &&
+        props.filteredYearSet.docCount > 0 && (
+          <div
+            style={{
+              width: props.width,
+              height: props.filterHeight,
+              backgroundColor: "#3e7fff",
+              borderRight: "solid 1px #f9f9fa",
```
Comment (NohYujeong) [inline app/containers/filterContainer/yearRangeSlider/index.tsx]: It seems that you can separate these style property values by declaring them as variables.

File: app/containers/filterContainer/yearRangeSlider/index.tsx [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+const Column: React.FunctionComponent<{
+  width: string;
+  height: string;
+  active: boolean;
+  yearSet: YearSet;
+  isSelecting: boolean;
+  isSelectMode: boolean;
+  filteredYearSet: YearSet | undefined;
+  filterHeight: string;
+  destinationParams: GoToYearFilteredSearchResultPageParams;
```
Comment (NohYujeong) [inline app/containers/filterContainer/yearRangeSlider/index.tsx]: It seems that this type designation can be separated by an interface.

File: app/containers/filterContainer/yearRangeSlider/index.tsx (hunk 1 above)
Comment (NohYujeong) [reply app/containers/filterContainer/yearRangeSlider/index.tsx]: or `const {location, history} = props;`or `React.FunctionComponent<YearRangeSliderProps> = {location, history ...} =>`

Comment (TylorShin) [reply app/containers/filterContainer/yearRangeSlider/index.tsx]: Resolve.

File: app/containers/filterContainer/yearRangeSlider/index.tsx (hunk 2 above)
Comment (TylorShin) [reply app/containers/filterContainer/yearRangeSlider/index.tsx]: Not necessary.

**Pull request merged**